odoo/addons/base/models/ir_actions_report.py

```diff
@@ -501,7 +501,7 @@ def _get_report_from_name(self, report_name):
         return report_obj.with_context(context).sudo().search(conditions, limit=1)
 
     @api.model
-    def barcode(self, barcode_type, value, width=600, height=100, humanreadable=0, quiet=1, mask=None):
+    def barcode(self, barcode_type, value, width=600, height=100, humanreadable=0, quiet=1, mask=None, qr_level=None):
```

Comment:
Is such a specific naming really useful? Couldn't this be more vague so it can be used by other barcodes with configurable error correction? (assuming that exists, I've no idea).

Reply:
it's also a public API change in stable. Maybe it's time to take a `barcode_options` dict, so we stop breaking that API every 2 months?

Reply:
The naming is specific because the parameter only has effect for QR codes.
ref: https://github.com/MatthewWilkes/reportlab/blob/master/src/reportlab/graphics/barcode/qr.py#L52

Also 'humanreadable' and 'quiet' look like they're used only for some kind of barcodes (and not for QRcodes).

Ref: https://github.com/MatthewWilkes/reportlab/blob/ba223fd973db95cdc96839a8024bbf2388a531d5/src/reportlab/graphics/barcode/widgets.py#L107

I'm not in my territory here, if you want my 2 humble cents, all parameters should fall in the `barcode_options` dict, because parameters change depending on the barcode type, properly mimicking the `createBarcodeDrawing` function.

I didn't do that because it's more a refactor than a bugfix in stable.